src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -192,12 +192,23 @@ class GeometricSpace(Entity):
     def get_geometric_space_for_atomic_cell(self, logger: 'BoundLogger') -> None:
         """
         Get the real space parameters for the atomic cell using ASE.
+        to_ase_atoms live under the parent ModelSystem.
 
         Args:
             logger (BoundLogger): The logger to log messages.
         """
+        parent = getattr(self, 'm_parent', None)
+        if parent is None or not hasattr(parent, 'to_ase_atoms'):
```

Comment:
I suggest not to use `hasattr` try to determine type of parent with `isinstance`